src/Abstractions/NexusMods.Abstractions.Loadouts.Synchronizers/ALoadoutSynchronizer.cs

```diff
@@ -123,14 +130,14 @@ public ValueTask<FileTree> FlattenedLoadoutToFileTree(FlattenedLoadout flattened
 
 
     /// <inheritdoc />
-    public async Task<DiskStateTree> FileTreeToDisk(FileTree fileTree, Loadout loadout, FlattenedLoadout flattenedLoadout, DiskStateTree prevState, GameInstallation installation, bool skipIngest = false)
+    public async Task<DiskStateTree> FileTreeToDisk(FileTree fileTree, Loadout? loadout, FlattenedLoadout flattenedLoadout, DiskStateTree prevState, GameInstallation installation, bool skipIngest = false)
     {
         // Return the new tree
         return await FileTreeToDiskImpl(fileTree, loadout, flattenedLoadout, prevState, installation, true);
     }
     
     [MethodImpl(MethodImplOptions.AggressiveInlining)]
-    internal async Task<DiskStateTree> FileTreeToDiskImpl(FileTree fileTree, Loadout loadout, FlattenedLoadout flattenedLoadout, DiskStateTree prevState, GameInstallation installation, bool fixFileMode, bool skipIngest = false)
+    internal async Task<DiskStateTree> FileTreeToDiskImpl(FileTree fileTree, Loadout? loadout, FlattenedLoadout flattenedLoadout, DiskStateTree prevState, GameInstallation installation, bool fixFileMode, bool skipIngest = false)
```

Comment:
Why did you make loadout nullable?

Reply:
`ResetToInitialState` calls the API without a Loadout when reverting the game folder to its original state to help with code reuse.

The Loadout here is used as a parameter to the APIs used for creating `GeneratedFile`(s) in `FileTreeToDiskImpl.

In the case of reverting the folder to stock, there (naturally) can't be generated files, as the original disk state is all static files. So I made the parameter nullable and added a defensive throw if there are generated files and loadout is null down below.